### ISSUE TITLE
[a11y] Search field not accessible

### ISSUE THREAD

**Issue by g123k** [openfoodfacts/openfoodfacts-server#10291]
Hi everyone,

To improve accessibility on the website, we have some "simple" fixes to implement.
We have two issues with the search field on the homepage:

- [ ] The `input` element should contain a `label` or an `aria-label`
- [ ] The `form`element should include a `role` to `search`
